tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/tasks/KonanBaseTasks.kt

```diff
@@ -113,7 +116,17 @@ abstract class KonanArtifactTask: KonanTargetableTask(), KonanArtifactSpec {
             val objectFactory = project.objects
             val linkUsage = objectFactory.named(Usage::class.java, Usage.NATIVE_LINK)
             val konanSoftwareComponent = config.mainVariant
-            konanSoftwareComponent.addVariant(NativeVariant(Names.of("${artifact.name.removeSuffix("$artifactSuffix")}_${target.name}"), linkUsage, null, linkUsage, platformConfiguration))
+            val variantName = "${artifact.name.removeSuffix("$artifactSuffix")}_${target.name}"
+            konanSoftwareComponent.addVariant(NativeVariantIdentity(
+                    variantName,
+                    project.provider{artifactName},
+                    project.provider{project.group.toString()},
+                    project.provider{ project.version.toString() },
+                    false,
```

Comment:
We can determine if the artifact is debuggable/optimized for an artifact produced by the compiler (the `konanCompileTask` and its child tasks). I think we need to take this into account while creating a `NativeVariantIdentity`.